olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PartitionsRequest.h

```diff
@@ -41,6 +42,22 @@ class DATASERVICE_READ_API PartitionsRequest final {
   /// The alias type of the vector of partitions ids
   using PartitionIds = std::vector<std::string>;
 
+  /// Additional field to request partition data size, see GetPartitions
+  static constexpr const char* kDataSize = "dataSize";
+
+  /// Additional field to request partition checksum, see GetPartitions
+  static constexpr const char* kChecksum = "checksum";
+
+  /// Additional field to request partition compressed data size, see
+  /// GetPartitions
+  static constexpr const char* kCompressedDataSize = "compressedDataSize";
+
+  /// Additional field to request partition crc, see GetPartitions
+  static constexpr const char* kCrc = "crc";
+
+  /// The alias type of the set of additional fields
+  using AdditionalFields = std::set<std::string>;
```

Comment:
I think we can safely use a vector. Let it be a user error if he specifies duplicate fields.